cli/module_graph2.rs

```diff
@@ -421,6 +439,128 @@ impl Graph2 {
     }
   }
 
+  fn get_info(
+    &self,
+    specifier: &ModuleSpecifier,
+    seen: &mut HashSet<ModuleSpecifier>,
+    totals: &mut HashMap<ModuleSpecifier, usize>,
+  ) -> ModuleInfo {
+    let not_seen = seen.insert(specifier.clone());
+    let module = self.modules.get(specifier).unwrap();
+    let mut deps = Vec::new();
+    let mut total_size = None;
+
+    if not_seen {
+      let mut seen_deps = HashSet::new();
+      // Check if the module has a type only dependency itself (e.g.
+      // `X-TypeScript-Types` header or triple slash ref)
+      if let Some((_, dep)) = &module.maybe_types {
+        if seen_deps.insert(dep.clone()) {
```

Comment:
I feel like type only dependencies should be separated in the JSON (type_deps field).

Reply:
If we are open to changing the JSON, I would like to rename quite a few fields actually, more aligned to the internals and consistent naming of things, as I think it really impedes maintainability for the long term.

Reply:
I guess changing the behaviour is outside of the scope of this PR.  I opened #7927 which we can deal with post this PR.